src/nvim/lua/executor.c

```diff
@@ -1063,6 +1063,16 @@ Object nlua_exec(const String str, const Array args, Error *err)
   return nlua_pop_Object(lstate, false, err);
 }
 
+bool nlua_ref_is_function(LuaRef ref)
+{
+  lua_State *const lstate = global_lstate;
+  nlua_pushref(lstate, ref);
+  bool is_function = (lua_type(lstate, -1) == LUA_TFUNCTION);
```

Comment:
hmm. Ideally we should support a callable table as well, but it is not critical for the first merge, so add it as a `TODO` comment here.